aptos-move/framework/aptos-framework/sources/object.spec.move

```diff
@@ -250,18 +431,27 @@ spec aptos_framework::object {
 
     spec ungated_transfer_allowed<T: key>(object: Object<T>): bool {
         aborts_if !exists<ObjectCore>(object.inner);
+        ensures result == global<ObjectCore>(object.inner).allow_ungated_transfer;
     }
 
     spec is_owner<T: key>(object: Object<T>, owner: address): bool{
         aborts_if !exists<ObjectCore>(object.inner);
+        ensures result == (global<ObjectCore>(object.inner).owner == owner);
     }
 
     spec owner<T: key>(object: Object<T>): address{
         aborts_if !exists<ObjectCore>(object.inner);
+        ensures result == global<ObjectCore>(object.inner).owner;
     }
 
     spec owns<T: key>(object: Object<T>, owner: address): bool {
+        // property 3: The 'indirect' owner of an object may transfer the object.
```

Comment:
I don't think this aborts_if condition encodes the property 3.

Reply:
property 3 could be proved by ensuring: when the owner is indirect, the transfer will still be successful.